openmdao/approximation_schemes/finite_difference.py

```diff
@@ -160,15 +160,16 @@ def compute_approximations(self, system, jac=None, deriv_type='partial'):
             One of 'total' or 'partial', indicating if total or partial derivatives are
             being approximated.
         """
+        if len(self._exec_list) == 0:
+            return
+
         if jac is None:
             jac = system._jacobian
 
         if deriv_type == 'total':
             current_vec = system._outputs
         elif deriv_type == 'partial':
             current_vec = system._residuals
-        else:
-            raise ValueError('deriv_type must be one of "total" or "partial"')
```

Comment:
I disagree, but don't feel like fighting it. I will put them back in if you put up a story for writing tests that cover all 4 of these "else" branches.

Reply:
you disagree that this leaves a hole in the logic and that the code is thus less robust?  an alternative would be to change the `elif` to an `else`, which closes the logic hole and just means you do partial if the arg is anything besides `total` (correctly spelled ;))

Reply:
I disagree that we need to check the inputs on deep private functions.  We don't do an isinstance on jac to make sure it is of class Jacobian or None.  Typoing "total" to "ttoal" will still raise an exception.  We won't get that exception if we enclose it in else: we will get partials instead of totals.

Reply:
I went with your second suggestion though. It is only called for total in one place.